File: practica1/P1S2/Java/src/practica1dss2/Cliente.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package practica1dss2;

/**
 *
 * @author USUARIO
 */
public class Cliente {
    
    public Cliente(){
        
    }
    
    public void crearProducto(FactoriaProducto factoria, int num){
        //Comida comida = factoria.crearComida();
        Tienda tienda = factoria.crearTienda(num);
        tienda.start();
    }
}

File: practica1/P1S2/Ruby/lib/Cliente.rb
require('./Comida.rb');
require('./Tienda.rb');

public
class Cliente

	public
	def initialize()

		
	end
  
  def crearProducto(factoria, num)
    tienda = factoria.crearTienda(num)
    Thread.new do tienda.run() end
  end
end
